### PR TITLE
Change web-platform-compat to browsercompat in documentation

### PR DESCRIPTION
**Pull request by jwhitlock**
Update text and links to change:
* Links to https://github.com/mozilla/web-platform-compat to https://github.com/mdn/browsercompat
* "Web Platform Compatibility API" to "Browser Compatibility API"
* Update badge URLs and use [shields.io](http://shields.io) for retina-ready images
* Move docs to https://browsercompat.readthedocs.org, and get RTD build working again
* Switch my email to Mozilla email

This still leaves the `webplatformcompat` app and `wpcsite` project folders, but I want to save those for a time when I can get a rapid code review, to minimize productivity losses.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -29,10 +29,6 @@ across languages and different presentations.
 .. _Mozilla Developer Network: https://developer.mozilla.org
 .. _CSS display property: https://developer.mozilla.org/en-US/docs/Web/CSS/display#Browser_compatibility
 
-*Note: This project will be renamed to browsercompat in the near future, to
-syncronize the project name with the planned production domain name
-https://browsercompat.org.*
-
```
Comment (groovecoder) [inline README.rst]: images duckduckgo com

File: requirements-rtd.txt [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+# TravisCI is now using ascii as system encoding, breaking setup.py
+git+git://github.com/erikrose/parsimonious@20863d86a#egg=parsimonious
+
+# Social loging support (Firefox Accounts)
+# python-openid >= 2.2.5  # Python 2.x - unused by app
+# python3-openid >= 3.0.1 # Python 3.x - unused by app
+requests==2.7.0
+oauthlib==0.7.2
+requests-oauthlib==0.5.0
+django-allauth==0.20.0
```
Comment (groovecoder) [inline requirements-rtd.txt]: Whoa, why all this? Because we use the app itself to generate the example API responses in the docs?

**Comment by groovecoder** [comment]
r+ after 1 question is answered.

File: requirements-rtd.txt (hunk 2 above)
Comment (jwhitlock) [inline requirements-rtd.txt]: Check out https://readthedocs.org for one of your products (kuma?).  There's a "build" tab that shows what gets done, which includes installing requirements.txt by default.

In docs/conf.py, I do import webplatformcompat (so I can grab the version) and setup the Django settings. That might be a leftover from when I was trying to use the sphinx autodocs, which are awful for Django projects.

I might be able to get by with a much smaller requirements.txt, but I was trying to get it to working from not working.  I'll give it a try,

**Comment by jwhitlock** [comment]
Rebased, with much smaller requirements file.

**Pull request merged**